plenum/server/primary_selector.py

```diff
@@ -82,15 +93,73 @@ def _is_master_instance(self, instance_id):
         # Instance 0 is always master
         return instance_id == 0
 
+    def _processCurrentStateMessage(self,
+                                    msg: CurrentState,
+                                    sender: str) -> bool:
+        """
+        Processes CurrentState messages. Once f+1 messages received
+        decides on a primary for specific replica.
+
+        It is a special case of primary selection in which current node
+        does not take part, but notified about.
+
+        :param msg: CurrentState message
+        :param sender: the name of the node which this message came from
+        """
+        logger.debug("{}'s primary selector started processing of "
+                     "CurrentState msg from {} : {}"
+                     .format(self.name, sender, msg))
+
+        view_no = msg.viewNo
+        if self.viewNo > view_no:
+            self.discard(msg,
+                         '{} got CurrentState from {} for view no {} '
+                         'whereas current view no is {}'
+                         .format(self, sender, view_no, self.viewNo),
+                         logMethod=logger.warning)
+            return False
+
+        view_change_dones = self._extract_view_change_dones(msg)
+        if len(view_change_dones) == 0:
+            logger.debug("{} ignoring CurrentState from {} since it "
+                         "brought no ViewChangeDone messages"
+                         .format(self, sender))
+            return False
+
+        the_only_view_change_done = view_change_dones[0]
+        self._track_current_state_message(sender,
+                                          view_no,
+                                          the_only_view_change_done)
+        self._startCurrentStateSelection(view_no)
+        return True
+
+    def _track_current_state_message(self, sender, view_no, view_change_dones):
```

Comment:
It expects a single ViewChangeDone, so why then the argument name is `view_change_doneS`